scripts/bench.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/python
+import argparse
+import os
+import sys
+
+from monotonic import monotonic
+
+from detect_secrets.core.usage import PluginOptions
+
+if os.name == 'posix' and sys.version_info[0] < 3:
+    import subprocess32 as subprocess
+else:
+    import subprocess
+
+
+def getArguments():
+    parser = argparse.ArgumentParser(description='Run some benchmarks.')
+    parser.add_argument(
+        'rest_args',
+        nargs=argparse.REMAINDER,
+        help='filenames to check or detect-secrets compatible arguments',
+    )
+    args = parser.parse_args()
+    return ['../.'] if len(args.rest_args) == 0 else args.rest_args
+
+
+def runBench(flags, arguments):
+    startTime = monotonic()
+    subprocess.call('detect-secrets scan'.split() + arguments + flags, stdout=subprocess.DEVNULL)
+    return monotonic() - startTime
+
+
+def printLine(name, time):
+    print('{:<20s}{:>20s}s'.format(name, str(time)))
+
+
+flags = list()
+aditionalArguments = getArguments()
+
+# Fill flags list with PluginOptions flag text
+for c, value in enumerate(PluginOptions.all_plugins, 0):
+    flags.append(value.disable_flag_text)
+
+# Run bechmarks for all the cases
+for c, value in enumerate(flags, 0):
+    temp = list(flags)
+    temp.pop(c)
+    if c == 0:
+        print('Scanning: ' + str(aditionalArguments))
+        print('-' * 42)
+        print('{:<20s}{:>20s}'.format('benchmark', 'time'))
+        print('-' * 42)
+        benchTime = runBench([], aditionalArguments)
+        printLine('all-plugins', runBench([], aditionalArguments))
+    printLine(value[5:], runBench(temp, aditionalArguments))
```

Comment:
Where did this `5` value come from?

Reply:
It crops the original value.disable_flag_text for removing the `--no-` part. e.g. from `--no-basic-auth-scan` to `basic-auth-scan`. Maybe one cleaner way to do this is instead of slicing the value.disable_flag_text using the value.classname! What do you think?

Reply:
Gotcha. Then, a better way would be to do:

```
value[len('--no-'):]
```

Reply:
Fixed!